### DISCUSSION TITLE
Auth0 integration is throwing an invalid token

### DISCUSSION THREAD

**Discussion by NicolasBonet** [Budibase/budibase#8126] [Help] [answered]
Hello guys!

I've followed this guide to do an Auth0 integration with Budibase:

https://docs.budibase.com/docs/sso-with-auth0

I have checked 3 now the entire config and our app looks ok, our Budibase stuff as well; however, when we try to call an endpoint with our token we get an invalid token, and endpoints respond: `token contains an invalid number of segments` and when we try to decode it we get the following error:

![image](https://user-images.githubusercontent.com/13696500/193880574-6c3abf81-8eee-4fc3-ae4f-2bb9a770fe42.png)

When I check the token the payload is missing (there are two dots, and nothing in between), do you guys know what could be the reason for this to happen?

Thanks :)

**Comment by mjashanks** [comment, marked as answer]
I can reproduce this also. To allow this to work properly, we would need to add the ability to set an "Audience" (see https://community.auth0.com/t/empty-payload-in-access-token/75952/3)

[This comment](https://github.com/Budibase/budibase/issues/4139#issuecomment-1160303972) suggests that it's possible to get a full token using the [Get User Info](https://auth0.com/docs/api/authentication#get-user-info) endpoint.

I have tried this. The request is successful, but it's just user-info - I can't see any tokens to be used.

Logged as an issue #8139

**Reply by NicolasBonet** [reply]
Thank you very much!

**Reply by aderici** [reply]
Also commented on the other issue but you can simply set a default audience on your auth0 dashboard under settings tab and you won't see this problem anymore

**Reply by NicolasBonet** [reply]
That did the trick, thank you!
